src/reconcile/rec_write.c

```diff
@@ -2239,16 +2239,34 @@ __wt_split_page_size(WT_BTREE *btree, uint32_t maxpagesize)
 }
 
 /*
- * __rec_min_split_page_size --
- *	Minimum split size boundary calculation: To track a boundary at the
- *	minimum split size that we could have split at instead of splitting at
- *	the split page size.
+ * __rec_split_page_size --
+ *	Split page size calculation: we don't want to repeatedly split every
+ *	time a new entry is added, so we split to a smaller-than-maximum page
+ *	size. Internal version that also adjusts based on standard compression.
  */
 static uint32_t
-__rec_min_split_page_size(WT_BTREE *btree, uint32_t maxpagesize)
+__rec_split_page_size(
+    WT_RECONCILE *r, WT_BTREE *btree, int split_pct, uint32_t maxpagesize)
 {
-	return (__rec_split_page_size_from_pct(
-	    WT_BTREE_MIN_SPLIT_PCT, maxpagesize, btree->allocsize));
+	uint32_t compadjust, split_size;
+
+	split_size = __rec_split_page_size_from_pct(
+	    split_pct, maxpagesize, btree->allocsize);
+
+	/*
+	 * Adjust for compression. Compression can be too effective, bound the
+	 * page split size at the maximum in-memory page image size, so point
+	 * updates don't cause forced eviction.
+	 */
+	compadjust = WT_PAGE_IS_INTERNAL(r->page) ?
+	    btree->intl_compadjust : btree->leaf_compadjust;
+	if (compadjust != 0) {
+		split_size = (split_size * compadjust) / WT_COMPRESS_ADJ;
```

Comment:
I don't really get this calculation. I didn't expect it to be divided by `WT_COMPRESS_ADJ`, oh is that just to avoid using a floating point number? i.e: we are shifting two decimal places?

I'd assumed that `WT_COMPRESS_ADJ` was used to control how much history weighted in when adjusting the compression rate.

Reply:
`WT_COMPRESS_ADJ` is just to avoid using a floating point number, I've renamed it to `WT_COMPRESS_SHIFT` and added a comment.